homeassistant/components/alarm_control_panel/__init__.py

```diff
@@ -203,19 +201,13 @@ def _report_deprecated_alarm_state_handling(self) -> None:
 
         Integrations should implement alarm_state instead of using state directly.
         """
-        self.__alarm_legacy_state_reported = True
-        if "custom_components" in type(self).__module__:
-            # Do not report on core integrations as they have been fixed.
-            report_issue = "report it to the custom integration author."
-            _LOGGER.warning(
-                "Entity %s (%s) is setting state directly"
-                " which will stop working in HA Core 2025.11."
-                " Entities should implement the 'alarm_state' property and"
-                " return its state using the AlarmControlPanelState enum, please %s",
-                self.entity_id,
-                type(self),
-                report_issue,
-            )
+        report_usage(
+            "is setting state directly which will stop working in HA Core 2025.11."
+            f" Entity {self.entity_id} ({type(self)}) should implement the 'alarm_state'"
+            " property and return its state using the AlarmControlPanelState enum.",
+            core_integration_behavior=ReportBehavior.LOG,
+            custom_integration_behavior=ReportBehavior.LOG,
```

Comment:
"LOG" is the default - so I'm not sure it is needed here.

```suggestion
```

If you want different behavior with core and custom integration, you need to use "mock_integration_frame" fixture + "integration_frame_path" parametrization

Reply:
Actually I will change this back as for core it should break as for core all integrations are fixed.

Reply:
I think if you have different behavior for core integration and custom integration, it's a good idea to have both set explicitly (even if it's the default for custom)
```suggestion
            core_integration_behavior=ReportBehavior.ERROR,
            custom_integration_behavior=ReportBehavior.LOG,
```